### PR TITLE
python3Packages.apptools: 4.5.0 -> 5.1.0

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Fix for py39: https://hydra.nixos.org/job/nixpkgs/trunk/python39Packages.apptools.x86_64-linux

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117696` at db6bb6b4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/e1cf06bdc1d2917239cf849c22e97350">python38Packages.envisage</a></li>
    <li><a href="https://gist.github.com/3782c3805e17b4fcc3629704afe38bdd">python39Packages.envisage</a></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.mayavi</li>
    <li>python39Packages.mayavi</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.apptools</li>
    <li>python39Packages.apptools</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/apptools/default.nix:1:39:
```
  |
1 | { lib, fetchPypi, buildPythonPackage, fetchpatch
  |                                       ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/development/python-modules/apptools/default.nix [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-  patches = [ (fetchpatch {
-      url = "https://github.com/enthought/apptools/commit/3734289d1a0ebd8513fa67f75288add31ed0113c.patch";
-      sha256 = "001012q1ib5cbib3nq1alh9ckzj588bfrywr8brkd1f6y1pgvngk";
-    })
+  propagatedBuildInputs = [
+    configobj
+    six
+    traitsui
+    importlib-resources
   ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/apptools/default.nix:21]:
```suggestion
  ] ++ lib.optionals (pythonOlder "3.9") [
    importlib-resources
  ];
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117696` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.mayavi: log was empty</li>
<li>python39Packages.apptools: log was empty</li>
<li>python39Packages.mayavi: log was empty</li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python38Packages.apptools</li>
</ul>
</details>

**Comment by rmcgibbo** [comment]
I'll bump envisage up to it's latest release.

**Pull request merged**